distributor/src/dist_controller/results.bal

```diff
@@ -135,36 +136,73 @@ service receiveResults on resultsListener {
     }
 }
 
+@http:ServiceConfig {
+    basePath: "/"
+}
+service workerReg on workerListener {
+    
+    @http:ResourceConfig {
+        webSocketUpgrade: {
+            upgradePath: "/workerRegister",
+            upgradeService: workerService
+        }
+    }
+    resource function upgrader(http:Caller caller, http:Request req) {
+        http:WebSocketCaller|http:WebSocketError wsEp = caller->acceptWebSocketUpgrade({});
+        if (wsEp is error) {
+            log:printError("Error occurred during WebSocket upgrade", wsEp);
+        }
+    }
+}
+
 function triggerAwaitNotification(string message) {
-    http:Client cl = <http:Client> secondaryDistributor;
-    var res = cl->post("/await", message);
+    string[] keys = workers.keys();
+    foreach string k in keys {
+        http:WebSocketCaller? cl = workers[k];
+        if cl is http:WebSocketCaller {
+            triggerAwaitNotificationByClient(cl, "\"" + message + "\"");
+        }
+    }
+}
+
+function triggerAwaitNotificationByClient(http:WebSocketCaller cl, string message) {
```

Comment:
Shall we move this function content to the parent function?

Reply:
This was changed as discussed offline.